clients/instance/ibm-pi-placement-groups.go

```diff
@@ -13,83 +16,107 @@ import (
+func (f *IBMPIPlacementGroupClient) Delete(id string) error {
+	params := p_cloud_placement_groups.NewPcloudPlacementgroupsDeleteParams().
+		WithContext(f.ctx).WithTimeout(helpers.PIDeleteTimeOut).
+		WithCloudInstanceID(f.cloudInstanceID).WithPlacementGroupID(id)
+	_, err := f.session.Power.PCloudPlacementGroups.PcloudPlacementgroupsDelete(params, f.authInfo)
 	if err != nil {
 		return fmt.Errorf(errors.DeletePlacementGroupOperationFailed, id, err)
 	}
 	return nil
 }
 
 // Adding a member to a  Placement Group
-func (f *IBMPIPlacementGroupClient) Update(placementdef *p_cloud_placement_groups.PcloudPlacementgroupsMembersPostParams, placementgroupid, powerinstanceid string) (*models.PlacementGroup, error) {
-
-	params := p_cloud_placement_groups.NewPcloudPlacementgroupsMembersPostParamsWithTimeout(helpers.PICreateTimeOut).WithCloudInstanceID(powerinstanceid).WithPlacementGroupID(placementgroupid).WithBody(placementdef.Body)
-	resp, err := f.session.Power.PCloudPlacementGroups.PcloudPlacementgroupsMembersPost(params, ibmpisession.NewAuth(f.session, f.powerinstanceid))
-
+func (f *IBMPIPlacementGroupClient) AddMember(id string, body *models.PlacementGroupServer) (*models.PlacementGroup, error) {
+	params := p_cloud_placement_groups.NewPcloudPlacementgroupsMembersPostParams().
+		WithContext(f.ctx).WithTimeout(helpers.PICreateTimeOut).
+		WithCloudInstanceID(f.cloudInstanceID).WithPlacementGroupID(id).
+		WithBody(body)
+	postok, err := f.session.Power.PCloudPlacementGroups.PcloudPlacementgroupsMembersPost(params, f.authInfo)
 	if err != nil {
-		return nil, fmt.Errorf(errors.UpdatePlacementGroupOperationFailed, powerinstanceid, placementgroupid, err)
+		return nil, fmt.Errorf(errors.AddMemberPlacementGroupOperationFailed, *body.ID, id, err)
 	}
-	return resp.Payload, nil
+	if postok == nil || postok.Payload == nil {
+		return nil, fmt.Errorf("failed to Add Member for instance %s and placement group %s", *body.ID, id)
+	}
+	return postok.Payload, nil
 }
 
 // Delete Member from Placement Group
-func (f *IBMPIPlacementGroupClient) DeleteMember(placementdef *p_cloud_placement_groups.PcloudPlacementgroupsMembersPostParams, placementgroupid, powerinstanceid string) (*models.PlacementGroup, error) {
-
-	params := p_cloud_placement_groups.NewPcloudPlacementgroupsMembersDeleteParamsWithTimeout(helpers.PICreateTimeOut).WithCloudInstanceID(powerinstanceid).WithPlacementGroupID(placementgroupid).WithBody(placementdef.Body)
-	resp, err := f.session.Power.PCloudPlacementGroups.PcloudPlacementgroupsMembersDelete(params, ibmpisession.NewAuth(f.session, f.powerinstanceid))
-
+func (f *IBMPIPlacementGroupClient) DeleteMember(id string, body *models.PlacementGroupServer) (*models.PlacementGroup, error) {
+	params := p_cloud_placement_groups.NewPcloudPlacementgroupsMembersDeleteParams().
+		WithContext(f.ctx).WithTimeout(helpers.PICreateTimeOut).
```

Comment:
In ibm-pi-placement-groups.go line 111 do we want to use helpers.PIDeleteTimeOut instead of helpers.PICreateTimeOut?

Reply:
it is fixed now